src/store/modules/app.js

```diff
@@ -148,20 +199,57 @@ const actions = {
         return resp;
       });
   },
+  saveConfiguratorSettings({ state }) {
+    localStorageSet(
+      CONSTS.configuratorSettings,
+      JSON.stringify(state.configuratorSettings)
+    );
+  },
   // if init state we just load and not toggling
-  toggleDarkMode({ commit, state }, init) {
-    let darkStatus = state.darkmodeEnabled;
+  toggleDarkMode({ commit, state, dispatch }, init) {
+    let darkStatus = state.configuratorSettings.darkmodeEnabled;
     if (!init) {
       darkStatus = !darkStatus;
     }
     if (darkStatus) {
-      localStorageSet('darkmode', '1');
       document.getElementsByTagName('html')[0].dataset.theme = 'dark';
     } else {
-      localStorageSet('darkmode', '0');
       document.getElementsByTagName('html')[0].dataset.theme = '';
     }
     commit('setDarkmode', darkStatus);
+    dispatch('saveConfiguratorSettings');
+  },
+  setFavoriteKeyboard({ commit, dispatch }, keyboard) {
+    commit('setFavoriteKeyboard', keyboard);
+    dispatch('saveConfiguratorSettings');
+  },
+  async loadApplicationState({ commit, dispatch }) {
+    console.log('loadApplicationState Start');
+    await dispatch('fetchKeyboards');
+    await dispatch('loadFavoriteKeyboard');
+    dispatch('toggleDarkMode', true);
+    console.log('loadApplicationState End');
+    commit('setAppInitialized', true);
```

Comment:
do we need a try catch and a return ?

Reply:
try catch would be overkill here as it's already catch in nested functions. What do you want to return? It's a void.

Reply:
I still think returning the dispatch would be better from a chaining point of view, but this is fine.